extensions/notification/queueProcessor/QueueProcessor.js

```diff
@@ -28,23 +30,35 @@ class QueueProcessor extends EventEmitter {
      *   consumer group ID
      * @param {String} notifConfig.queueProcessor.retryTimeoutS -
      *   number of seconds before giving up retries of an entry
-     * @param {Object} destinationConfig - destination configuration object
      * @param {String} destinationConfig.type - destination type
      * @param {String} destinationConfig.host - destination host
      * @param {String} destinationConfig.auth - destination auth configuration
      * @param {String} destinationId - resource name/id of destination
+     * @param {Object} destinationAuth - destination authentication config
      */
-    constructor(mongoConfig, kafkaConfig, notifConfig, destinationConfig,
-        destinationId) {
+    constructor(mongoConfig, kafkaConfig, notifConfig, destinationId,
+        destinationAuth) {
         super();
         this.mongoConfig = mongoConfig;
         this.kafkaConfig = kafkaConfig;
         this.notifConfig = notifConfig;
-        this.destinationConfig = destinationConfig;
         this.destinationId = destinationId;
+        this.destinationConfig
+            = notifConfig.destinations.find(dest => dest.resource === destinationId);
+        assert(this.destinationConfig, 'Invalid destination argument. Destination ' +
+        'could not be found in destinations defined');
```

Comment:
I think we can remove the first sentence of invalid argument and just say it could not be found. But do also include what our ID was so we know which one wasn't found.